### PR TITLE
feat: add `constants/float32/max-safe-nth-lucas`

### LINKED ISSUE #3336: [RFC]: add `constants/float32/max-safe-nth-lucas`
### Description

This RFC proposes adding `constants/float32/max-safe-nth-lucas`.

```javascript
var bool = ( FLOAT32_MAX_SAFE_NTH_LUCAS === 34 );
// returns true
```

### Related Issues

None.

### Questions

No.

### Other

No.

### Checklist

- [x] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [x] Searched for existing issues and pull requests.
- [x] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by aayush0325**
Resolves #3336 

## Description

> What is the purpose of this pull request?

This pull request:

-   adds `constants/float32/max-safe-nth-lucas`

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #3336 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

**Comment by aayush0325** [comment]
I arrived at this value using the following code 

```javascript
function lucas(n) {
    if (n < 0) return null;
    if (n === 0) return 2;
    if (n === 1) return 1;
    
    let a = 2;
    let b = 1;
    let c;
    
    for (let i = 2; i <= n; i++) {
        c = a + b;
        a = b;
        b = c;
    }
    return b;
}

const MAX_SAFE_FLOAT32 = 16777215; // taken from @stdlib/constants/float32/max-safe-integer

for (let n = 0; n < 100; n++) {
    const lucasValue = lucas(n);
    
    if (lucasValue > MAX_SAFE_FLOAT32) {
        console.log(`Max safe n: ${n-1}`);
        console.log(`Last safe Lucas value: ${lucas(n-1)}`);
        console.log(`First unsafe Lucas value: ${lucasValue}`);
        break;
    }
}
```

which gave the output

```bash
Max safe n: 34
Last safe Lucas value: 12752043
First unsafe Lucas value: 20633239
```

File: lib/node_modules/@stdlib/constants/float32/max-safe-nth-lucas/README.md [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+    }
+    b = 1;
+    for ( i = 2; i <= n; i++ ) {
+        c = a + b;
+        a = b;
+        b = c;
+    }
+    return b;
+}
+
```
Comment (kgryte) [inline lib/node_modules/@stdlib/constants/float32/max-safe-nth-lucas/README.md]:
```suggestion
function lucas( n ) {
    var a;
    var b;
    var c;
    var i;

    a = 2;
    if ( n === 0 ) {
        return a;
    }
    b = 1;
    for ( i = 2; i <= n; i++ ) {
        c = a + b;
        a = b;
        b = c;
    }
    return b;
}

var v;
var i;
```

File: lib/node_modules/@stdlib/constants/float32/max-safe-nth-lucas/examples/index.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+	}
+	b = 1;
+	for ( i = 2; i <= n; i++ ) {
+		c = a + b;
+		a = b;
+		b = c;
+	}
+	return b;
+}
+
```
Comment (kgryte) [inline lib/node_modules/@stdlib/constants/float32/max-safe-nth-lucas/examples/index.js]:
```suggestion
function lucas( n ) {
	var a;
	var b;
	var c;
	var i;

	a = 2;
	if ( n === 0 ) {
		return a;
	}
	b = 1;
	for ( i = 2; i <= n; i++ ) {
		c = a + b;
		a = b;
		b = c;
	}
	return b;
}

var v;
var i;
```

**Review by kgryte — APPROVED** [review]
LGTM. Thanks, @aayush0325!

File: lib/node_modules/@stdlib/constants/float32/max-safe-nth-lucas/include/stdlib/constants/float32/max_safe_nth_lucas.h [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+* limitations under the License.
+*/
+
+#ifndef STDLIB_CONSTANTS_FLOAT32_MAX_SAFE_NTH_LUCAS_H
+#define STDLIB_CONSTANTS_FLOAT32_MAX_SAFE_NTH_LUCAS_H
+
+/**
+* Macro for the maximum safe nth Lucas number when stored in single-precision floating-point format.
+*/
+#define STDLIB_CONSTANT_FLOAT64_MAX_SAFE_NTH_LUCAS 34
```
Comment (aayush0325) [inline lib/node_modules/@stdlib/constants/float32/max-safe-nth-lucas/include/stdlib/constants/float32/max_safe_nth_lucas.h:25]: There is a typo here that must be fixed, should i reopen this PR and make a commit to fix this or should i make a new PR here @kgryte, I'm really sorry for this.

**Pull request merged**